.gitignore

```diff
@@ -10,3 +10,4 @@
 /tags
 __pycache__
 .tox/
+.idea
```

Comment:
Please avoid no-newline-at-end-of-file. If you're using sublime text please configure it to not do that[1].

It causes merge churn.  It might be worth rebasing to fix that and squash some of these later fixes into their original commit in this series too.

[1] https://forum.sublimetext.com/t/make-saving-newline-at-eof-the-installation-default/9842